File: rust_library/divisors.rs
fn div_list(x: usize) -> Vec<usize> {
    let mut div_l: Vec<usize> = Vec::new();
    let mut div_r: Vec<usize> = Vec::new();
    let mut i: usize = 1;
    while i * i <= x {
        if x % i == 0 {
            div_l.push(i);
            if i != x / i {
                div_r.push(x / i);
            }
        }
        i += 1;
    }
    // div_r.reverse();
    // div_l.append(&mut div_r);
    // div_l
    div_l.iter().cloned().chain(div_r.iter().rev().cloned()).collect()
}

File: cpp_library/divisors.cpp
vector<long long> div_list(long long x) {
    vector<long long> l, r;
    long long i = 1;
    while (i * i <= x) {
        if (x % i == 0) {
            l.push_back(i);
            if (i != x / i) {
                r.push_back(x / i);
            }
        }
        i++;
    }
    reverse(r.begin(), r.end());
    l.insert(l.end(), r.begin(), r.end());
    return l;
}
